lib/iris/fileformats/netcdf.py

```diff
@@ -35,8 +35,7 @@
 import warnings
 
 import biggus
-import iris.proxy
-iris.proxy.apply_proxy('netCDF4', globals())
+import netCDF4
```

Comment:
Surprisingly, I think we are "close" to this being an optional dependency.
Apart from the obvious fileformat modules we have no direct I think there are no essential dependencies on NetCDF4.

However, a lot of things do need netcdftime.
That means a lot of *other* file formats (like PP and ff-fast-structured).  There are some uses of this in the core modules too : coords, cube, unit (though that is now going), and util.
(note that 'util' is also core as other core modules like coords and cube import it).